### PR TITLE
feat: add ipfs URL to gateway URL converter

### PR DESCRIPTION
**Pull request by alanshaw**
Adds a function that converts `ipfs://bafy...` URLs to gateway URLs like `https://dweb.link/ipfs/bafy...`

resolves https://github.com/ipfs-shipyard/nft.storage/issues/158

### CODE REVIEW

File: packages/client/src/embed.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Convert an IPFS URL (starting ipfs://) to a gateway URL (starting https://)
+ * that can be embedded in a webpage. If the passed URL is not an IPFS URL it is
+ * returned as a new URL object with no further changes.
+ *
+ * @param {string|URL} url An IPFS URL e.g. ipfs://bafy.../path
+ * @param {EmbedURLOptions} [options] Options that allow customization of the gateway used.
+ * @returns {URL} An IPFS gateway URL e.g. https://dweb.link/ipfs/bafy.../path
+ */
+export const toEmbedURL = (url, options = {}) => {
```
Comment (olizilla) [inline packages/client/src/embed.js:17]: Minor: `toEmbedURL` implies you can't use an ipfs:// url to embed content. Perhaps `toHttpsUrl` or `toGatewayUrl`

File: packages/client/test/embed.spec.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+describe('toEmbedURL', () => {
+  it('converts ipfs:// URL to a gateway URL', () => {
+    const ipfsURL =
+      'ipfs://bafybeibcepxovpzwu6ug3ofoyejj3a3z5dqqn6invyiqg3jhuzr27ticp4/yesthisisdog.jpg'
+    const gwURL =
+      'https://dweb.link/ipfs/bafybeibcepxovpzwu6ug3ofoyejj3a3z5dqqn6invyiqg3jhuzr27ticp4/yesthisisdog.jpg'
+    const result = toEmbedURL(ipfsURL)
+    assert.equal(result.toString(), gwURL)
+  })
+
```
Comment (olizilla) [inline packages/client/test/embed.spec.js:13]: Minor: test for URL object as param

File: packages/client/package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "build:ts": "tsc --build",
     "build:cjs": "rollup --config rollup.config.js",
     "coverage": "nyc report --reporter=text-lcov > coverage.lcov && npx codecov",
-    "typedoc": "typedoc --entryPoints src  --out ../docs/client",
+    "typedoc": "typedoc --entryPoints src  --out ../../docs/client",
```
Comment (olizilla) [inline packages/client/package.json:34]: unrelated fix?

Comment (alanshaw) [reply packages/client/package.json:34]: Yes, a hangover from when we moved everything into the `packages` directory. These should be written to the root.

File: packages/client/src/embed.js (hunk 1 above)
Comment (alanshaw) [reply packages/client/src/embed.js:17]: I called it this for consistency with the `.embed()` function on the return value from the metadata API - `client.store()`.

I think `toGatewayUrl` is probably more descriptive.

@Gozala do you have an opinion?

**Review by hugomrdias — APPROVED** [review]
LGTM
